proxy/coco/src/ethereum_claim_ext.rs

```diff
@@ -0,0 +1,32 @@
+//! The user identity doc extension for Ethereum addresses attestation.
+//! See [the RFC](docs/ethereum_attestation.md).
+
+use lazy_static::lazy_static;
+use librad::identities::payload::HasNamespace;
+use serde::{Deserialize, Serialize};
+use url::Url;
+
+/// The user identity doc extension for Ethereum addresses claims.
+/// Meaningful only if confirmed on Ethereum, see [the RFC](docs/ethereum_attestation.md).
+#[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct V1 {
+    /// The Ethereum address claimed by the user, encoded according to
+    /// [EIP-55](https://eips.ethereum.org/EIPS/eip-55).
+    pub address: String,
+    /// The timestamp before which the address claim is valid, encoded according to
+    /// [JavaScript Date](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toJSON)
+    pub expiration: String,
```

Comment:
Why a `String` rather than a time type?

Reply:
I don't have a strong opinion and this may be a wrong approach, but: we're just passing around the value, we don't care about its content. The time type would be parsing this string only to format it, which would introduce new bug opportunities and new dependencies. The app would still needs to parse and validate the expiration date received from the API.

Reply:
Are you not going to check the expiration value at any point? At which point you'll need to compare times, not strings.

Reply:
I think that it's the App's responsibility to interpret the expiration value (e.g. to tell that it's already in the past) and only then comparisons will be needed. The only thing that api and coco can check is if the format is correct.

Anyway I thought about it and I'll add the validation. At the very least it's worth guarding the identity doc from storing junk. It's also easier to reason about the successfully parsed time than about "a string, which should follow this format, but it may not and it's still fine despite being wrong".

Reply:
Ya, not accepting junk is very valuable ;)

Reply:
Fixed